shopinvader_api_payment/routers/payment.py

```diff
@@ -0,0 +1,219 @@
+            tx_sudo,
+            payable=data.payable,
+            frontend_redirect_url=data.frontend_redirect_url,
+        )
+        return transaction_processing_values
+    else:
+        raise NotImplementedError("Only redirect flow is supported")
+
+
+class ShopinvaderApiPaymentRouterHelper(models.AbstractModel):
+    _name = "shopinvader_api_payment.payment_router.helper"
+    _description = "ShopInvader API Payment Router Helper"
+
+    def _get_additional_transaction_create_values(
+        self,
+        data: TransactionCreate,
+        odoo_env: Annotated[api.Environment, Depends(odoo_env)],
+    ) -> dict:
+        # Intended to be extended for invoices, carts...
+        additional_transaction_create_values = {}
+        return additional_transaction_create_values
+
+    def _get_tx_create_values(
+        self,
+        data: TransactionCreate,
+        provider_sudo: PaymentProvider,
+        odoo_env: Annotated[api.Environment, Depends(odoo_env)],
+    ) -> dict:
+        try:
+            payable_obj = Payable.decode(odoo_env, data.payable)
+        except Exception as e:
+            _logger.info("Could not decode payable")
+            raise HTTPException(403) from e
+        additional_transaction_create_values = (
+            self._get_additional_transaction_create_values(data, odoo_env)
+        )
+
+        is_validation = False  # future
+        # compute transaction reference from payable reference
+        tx_reference = (
+            odoo_env["payment.transaction"]
+            .sudo()
+            ._compute_reference(
+                provider_code=provider_sudo.code,
+                prefix=payable_obj.payable_reference,
+                # TODO are custom_create_values and kwargs really needed
+                # **(custom_create_values or {}),
+                # **kwargs
+            )
+        )
+
+        return {
+            "provider_id": data.provider_id,
+            "reference": tx_reference,
+            "amount": payable_obj.amount,
+            "currency_id": payable_obj.currency_id,
+            "partner_id": payable_obj.partner_id,
+            # 'token_id': token_id,
+            "operation": f"online_{data.flow}" if not is_validation else "validation",
+            "tokenize": False,
+            **additional_transaction_create_values,
+        }
+
+    def _create_transaction(
+        self,
+        data: TransactionCreate,
+        provider_sudo: PaymentProvider,
+        request: Request,
+        odoo_env: Annotated[api.Environment, Depends(odoo_env)],
+    ) -> dict:
+        transaction_values = odoo_env[
+            "shopinvader_api_payment.payment_router.helper"
+        ]._get_tx_create_values(data, provider_sudo, odoo_env)
+        tx_sudo = (
+            odoo_env["payment.transaction"]
+            .sudo()
+            .with_context(
+                shopinvader_api_payment=True,
+                shopinvader_api_payment_frontend_redirect_url=data.frontend_redirect_url,
```

Comment:
Why not creating a field on transaction, not all providers support sending back context.

(See https://github.com/akretion/odoo-shopinvader/commit/651b1bcc44ebe08bb3ac7573f23e7e776ad12f90 as I was in that case)